cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ReplicaDistributionGoal.java

```diff
@@ -121,9 +118,23 @@ public String name() {
   @Override
   protected void initGoalState(ClusterModel clusterModel, OptimizationOptions optimizationOptions) {
     super.initGoalState(clusterModel, optimizationOptions);
-    for (Broker broker : clusterModel.brokers()) {
-      BrokerAndSortedReplicas bas = new BrokerAndSortedReplicas(broker, broker.replicaComparator(DISK));
-      _brokerAndReplicasMap.put(broker.id(), bas);
+    clusterModel.trackSortedReplicas(name(),
+                                     ReplicaSortFunctionFactory.prioritizeImmigrants(),
```

Comment:
I am curious about the pros/cons of using `ReplicaSortFunctionFactory#prioritizeImmigrants()` with `SortedReplicas#sortedReplicas()` vs. `ReplicaSortFunctionFactory#deprioritizeImmigrants()` with `SortedReplicas#reverselySortedReplicas()` as in the other goals?

For example, in ResourceDistribution goal, I feel that reversely sorting the following would lead to an inconsistent order to regular sorting of the clustermodel in Replica distribution goal.

```
    clusterModel.trackSortedReplicas(sortName(),
                                     ReplicaSortFunctionFactory.deprioritizeImmigrants(),
                                     ReplicaSortFunctionFactory.sortByMetricGroupValue(resource().name()));
```

Reply:
In `ResourceDistributionGoal`/`ResourceCapacityGoal` we prioritize immigrant replicas and replica with **larger** size, we expect it will minimize **number of replica** to move.

In `ReplicaDistributionGoal` we prioritize immigrant replicas and replica with **smaller** size,  here we expect to minimize **amount of data** to move.